### PR TITLE
Feature/extend asset profile for currency

### LINKED ISSUE #3488: Extend asset profile for currency
As a follow-up to https://github.com/ghostfolio/ghostfolio/issues/3006, add the missing attributes for currencies (e.g. `USDEUR`) to market data endpoint `api/v1/admin/market-data/:dataSource/:symbol`:

* `activitiesCount`
* `dataSource`
* `dateOfFirstActivity`

asset-profile-usdeur

### PR DESCRIPTION
**Pull request by SirZemar**
Closes #3488

### CODE REVIEW

File: apps/api/src/app/admin/admin.service.ts [hunk 1]
```diff
@@ -295,6 +298,27 @@ export class AdminService {
     dataSource,
     symbol
   }: UniqueAsset): Promise<AdminMarketDataDetails> {
+    let activitiesCount: EnhancedSymbolProfile['activitiesCount'] = 0;
+    let dateOfFirstActivity: EnhancedSymbolProfile['dateOfFirstActivity'];
+    let currency: EnhancedSymbolProfile['currency'] = '-';
+
+    if (isCurrency(symbol.replace(DEFAULT_CURRENCY, ''))) {
+      currency = symbol.replace(DEFAULT_CURRENCY, '');
+      const { _count, _min } = await this.prismaService.order.aggregate({
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:307]: Can you extract this as a function, move it to `OrderService` and reuse it also in `getMarketDataForCurrencies()`?

File: apps/api/src/app/admin/admin.service.ts [hunk 2]
```diff
@@ -322,12 +346,14 @@ export class AdminService {
     return {
       marketData,
       assetProfile: assetProfile ?? {
+        currency,
+        dataSource,
         symbol,
-        currency: '-'
+        activitiesCount,
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:352]: Move up ⬆️

File: apps/api/src/app/admin/admin.service.ts [hunk 3]
```diff
@@ -322,12 +346,14 @@ export class AdminService {
...
+        dateOfFirstActivity
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:353]: Move up ⬆️

File: apps/api/src/app/admin/admin.service.ts [hunk 4]
```diff
@@ -322,12 +346,14 @@ export class AdminService {
...
       }
     };
   }
-
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:330]: ❓

File: apps/api/src/app/admin/admin.service.ts [hunk 5]
```diff
@@ -32,6 +34,7 @@ import {
   Property,
   SymbolProfile
 } from '@prisma/client';
+import { isNumber } from 'class-validator';
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:37]: ❓

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thanks, @SirZemar! 😎
I have added a few comments.

File: apps/api/src/app/order/order.service.ts [hunk 6]
```diff
@@ -541,4 +545,21 @@ export class OrderService {
       where
     });
   }
+
+  public async getCountAndDateMin(
```
Comment (dtslvr) [inline apps/api/src/app/order/order.service.ts:549]: Maybe `getStatisticsByCurrency`❓

File: apps/api/src/app/order/order.service.ts [hunk 7]
```diff
@@ -541,4 +545,21 @@ export class OrderService {
...
+    currency: EnhancedSymbolProfile['currency']
+  ): Promise<[number, Date]> {
```
Comment (dtslvr) [inline apps/api/src/app/order/order.service.ts:551]: `Promise<{activitiesCount: number; dateOfFirstActivity: Date}>`

File: apps/api/src/app/order/order.service.ts [hunk 8]
```diff
@@ -541,4 +545,21 @@ export class OrderService {
...
+      _min: {
+        date: true
+      },
+      where: { SymbolProfile: { currency } }
+    });
+    const activitiesCount: EnhancedSymbolProfile['activitiesCount'] =
+      _count as number;
+    const dateOfFirstActivity: EnhancedSymbolProfile['dateOfFirstActivity'] =
+      _min.date;
+    return [activitiesCount, dateOfFirstActivity];
```
Comment (dtslvr) [inline apps/api/src/app/order/order.service.ts:563]: `return { activitiesCount, dateOfFirstActivity };`

File: apps/api/src/app/admin/admin.service.ts [hunk 9]
```diff
@@ -346,14 +337,15 @@ export class AdminService {
     return {
       marketData,
       assetProfile: assetProfile ?? {
+        activitiesCount,
+        dateOfFirstActivity,
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:341]: Move down ⬇️

File: apps/api/src/app/admin/admin.service.ts [hunk 10]
```diff
@@ -466,11 +454,11 @@ export class AdminService {
             dataSource,
             marketDataItemCount,
             symbol,
-            activitiesCount: _count as number,
+            activitiesCount,
```
Comment (dtslvr) [inline apps/api/src/app/admin/admin.service.ts:457]: Move up ⬆️

**Review by dtslvr — CHANGES_REQUESTED** [review]
Looks better, thank you. Just a few additional minor improvements 🙂

File: apps/api/src/app/order/order.service.ts (hunk 6 above)
Comment (SirZemar) [reply apps/api/src/app/order/order.service.ts:549]: Definitely better. I can also suggest adding something that indicates where these stats are relevant, e.g., `getActivityStatsByCurrency`.

**Review by dtslvr — APPROVED** [review]
Thanks a lot for the adaptions, @SirZemar 👍🏻
I have tested it and slightly refactored the code. Ready to merge now.

**Pull request merged**